app/grandchallenge/algorithms/views.py

```diff
@@ -109,6 +184,45 @@ def form_valid(self, form):
         return response
 
 
+class PhaseAlgorithmCreate(
+    LoginRequiredMixin,
+    PhaseConfiguredForAlgorithmSubmissionMixin,
+    UserCanSubmitToPhaseMixin,
+    VerificationRequiredMixin,
+    CreateView,
+):
+    model = Algorithm
+    form_class = AlgorithmForPhaseForm
+
+    def form_valid(self, form):
+        response = super().form_valid(form=form)
+        self.object.add_editor(self.request.user)
+        self.object.logo = self.phase.challenge.logo
+        self.object.workstation_config = self.phase.workstation_config
+        self.object.hanging_protocol = self.phase.hanging_protocol
+        self.object.view_content = self.phase.view_content
+        self.object.display_editors = True
+        self.object.contact_email = self.request.user.email
+        self.object.modalities.set(self.phase.challenge.modalities.all())
+        self.object.structures.set(self.phase.challenge.structures.all())
+        self.object.inputs.set(self.phase.algorithm_inputs.all())
+        self.object.outputs.set(self.phase.algorithm_outputs.all())
+        if self.phase.workstation:
+            self.object.workstation = self.phase.workstation
+        self.object.save()
+        return response
```

Comment:
`form_valid` is only responsible for saving the form. You should only need to change this method if you need to add in things like signals. The form itself is responsible for setting any other parameters and doing the object validation. Probably what you need here is a separate form, with the inherited values from the phase set as non-editalble, hidden, fields.